src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState, useEffect } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import { Loader } from '../components/Loader';
+import { getPeople } from '../api';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { PeopleTable } from '../components/PeopleTable';
+import { Gender } from '../types/Gender';
+
+const MALE_GENDER = 'm';
+const FEMALE_GENDER = 'f';
+
+export const PeoplePage: React.FC = () => {
+  const [searchParams, setSearcParams] = useSearchParams();
+  const params = new URLSearchParams(searchParams);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [selectedPersonSlug, setSelectedPersonSlug] = useState<string>('');
+  const [filterGenderStatus, setFilterGenderStatus]
+  = useState<Gender>(Gender.ALL);
+  const [filteredPeople, setFilteredPeople] = useState<Person[]>([]);
+  const [selectedCenturies, setSelectedCenturies] = useState<string[]>([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const { slug } = useParams();
```

Comment:
I think this toooo much code for one component. Let's do following:
1. Let's put filtration to separate function in separate file instead of useEffect. This function should take `searchParams` object. Store the result of filtration in variable.
2. `query` handling should be in `PeopleFilters`

Why do you need all those states like `selectedPersonSlug`, `filterGenderStatus`, etc. This the whole point of using router, because your url parts are basically a state.

Reply:
yes, this topic is quite difficult for me, I realized the mistakes and will redo this task